py/py.mk

```diff
@@ -103,6 +103,13 @@ $(BUILD)/$(BTREE_DIR)/%.o: CFLAGS += -Wno-old-style-definition -Wno-sign-compare
 $(BUILD)/extmod/modbtree.o: CFLAGS += $(BTREE_DEFS)
 endif
 
+# External modules written in C.
+ifneq ($(MODULE_DIR),)
+CFLAGS_MOD += -DMICROPY_HAS_EXTRA_MODULES
```

Comment:
This line won't be necessary if you get rid of `MICROPY_HAS_EXTRA_MODULES` (as discussed above).